api/python/t4/api.py

```diff
@@ -101,6 +102,44 @@ def delete(target):
     delete_object(bucket, path)
 
 
+def delete_package(name, registry=None):
+    if not re.match(PACKAGE_NAME_FORMAT, name):
+        raise QuiltException("Invalid package name, must contain exactly one /.")
+
+    if name not in list_packages(registry):
+        raise QuiltException("No such package exists in the given directory.")
+
+    registry_base_path = get_package_registry(fix_url(registry) if registry else None)
+    registry_url = urlparse(registry_base_path)
+    if registry_url.scheme != 'file':
+        raise NotImplementedError
+
+    pkg_namespace, pkg_subname = name.split("/")
+
+    registry_dir = pathlib.Path(parse_file_url(registry_url)).as_posix()
```

Comment:
`pathlib.Path` is not needed here - `registry_dir` is already a `Path`. Same below.